fem/libceed/ceed.cpp

```diff
@@ -364,18 +455,32 @@ void CeedPAAssemble(const CeedPAOperator& op,
    // Create the operator that builds the quadrature data for the operator.
    CeedOperatorCreate(ceed, ceedData.build_qfunc, NULL, NULL,
                       &ceedData.build_oper);
-   if (ceedData.coeff_type==CeedCoeff::Grid)
+   switch (ceedData.coeff_type)
    {
-      CeedGridCoeff* ceedCoeff = (CeedGridCoeff*)ceedData.coeff;
-      InitCeedBasisAndRestriction(*ceedCoeff->coeff->FESpace(), irm, ceed,
-                                  &ceedCoeff->basis,
-                                  &ceedCoeff->restr);
-      CeedVectorCreate(ceed, ceedCoeff->coeff->FESpace()->GetNDofs(),
-                       &ceedCoeff->coeffVector);
-      CeedVectorSetArray(ceedCoeff->coeffVector, CEED_MEM_HOST, CEED_USE_POINTER,
-                         ceedCoeff->coeff->GetData());
-      CeedOperatorSetField(ceedData.build_oper, "coeff", ceedCoeff->restr,
-                           ceedCoeff->basis, ceedCoeff->coeffVector);
+      case CeedCoeff::Const:
+         break;
+      case CeedCoeff::Grid:
+      {
+         CeedGridCoeff* gridCoeff = (CeedGridCoeff*)ceedData.coeff;
+         InitCeedBasisAndRestriction(*gridCoeff->coeff->FESpace(), irm, ceed,
+                                     &gridCoeff->basis,
+                                     &gridCoeff->restr);
+         CeedOperatorSetField(ceedData.build_oper, "coeff", gridCoeff->restr,
+                              gridCoeff->basis, gridCoeff->coeffVector);
+      }
+      break;
+      case CeedCoeff::Quad:
+      {
+         CeedQuadCoeff* quadCoeff = (CeedQuadCoeff*)ceedData.coeff;
+         const int ncomp = 1;
+         CeedInt strides[3] = {1, nqpts, ncomp*nqpts};
+         CeedElemRestrictionCreateStrided(ceed, nelem, nqpts, ncomp,
+                                          nelem*ncomp*nqpts, strides,
+                                          &quadCoeff->restr);
+         CeedOperatorSetField(ceedData.build_oper, "coeff", quadCoeff->restr,
+                              CEED_BASIS_COLLOCATED, quadCoeff->coeffVector);
+      }
+      break;
```

Comment:
Should there be default case?

Reply:
All cases are treated, the compiler would issue a warning otherwise.

Reply:
I should remove the `error` in the other `switch`